src/typeahead/typeahead.spec.ts

```diff
@@ -194,6 +197,93 @@ describe('ngb-typeahead', () => {
+         const checkSearchCount = (expected, context) => {
+           expect(searchCount).toBe(expected, `Search count is not correct: ${context}`);
+           searchCount = 0;
+         };
+
+         const checkWindowIsClosed = () => {
+           expect(getWindow(compiled)).toBeNull();
+           expect(fixture.componentInstance.model).toBe(undefined);
+           expect(getNativeInput(compiled).value).toBe('');
+         };
+
+         const checkWindowIsOpen = () => { expect(getWindow(compiled)).not.toBeNull(); };
+
+         // focusing the input triggers a search and opens the dropdown
+         getNativeInput(compiled).focus();
+         // on IE the focus can be asynchronous, so we need to wait a tick before continuing
+         await new Promise(resolve => setTimeout(resolve, 0));
+         checkSearchCount(1, 'on first focus');
+         checkWindowIsOpen();
+
+         // clicking again in the input while the dropdown is open doesn't trigger a new search and keeps the dropdown
+         // open
+         getNativeInput(compiled).click();
+         checkSearchCount(0, 'on input click when dropdown already open');
+         checkWindowIsOpen();
+
+         // closing the dropdown but keeping focus
+         const event = createKeyDownEvent(Key.Escape);
+         getDebugInput(fixture.debugElement).triggerEventHandler('keydown', event);
+         fixture.detectChanges();
+         checkWindowIsClosed();
+
+         // clicking again in the input while already focused but dropdown closed triggers a search and opens the
+         // dropdown
+         getNativeInput(compiled).click();
+         checkSearchCount(1, 'on input click when input is already focused but dropdown is closed');
+         checkWindowIsOpen();
+
+         // closing the dropdown and losing focus
+         fixture.nativeElement.click();
+         getNativeInput(compiled).blur();
+         checkWindowIsClosed();
+
+         // Clicking directly, putting focus at the same time, triggers only one search and opens the dropdown
+         getNativeInput(compiled).click();
+         checkSearchCount(1, 'on input focus specifically with a click');
+         checkWindowIsOpen();
+       }));
+
+    it('should preserve value previously selected with mouse when reopening with focus then closing without selection',
+       () => {
+         const fixture = createTestComponent(`<input type="text" [ngbTypeahead]="find"/>`);
+         const compiled = fixture.nativeElement;
+
+         fixture.whenStable().then(() => {
+           // open with partial input
+           changeInput(compiled, 'o');
+           fixture.detectChanges();
+
+           // select with click
+           getWindowLinks(fixture.debugElement)[0].triggerEventHandler('click', {});
+           fixture.detectChanges();
+           expectInputValue(compiled, 'one');
+
+           // open again but with focus
+           getNativeInput(compiled).blur();
+           getNativeInput(compiled).focus();
```

Comment:
Could you please add expectation here to prove that a window with results is open? I don't see `(focus)` handler on this `<input>` field so I'm not sure what would trigger opening?

Reply:
Yes, it was there initially but I removed it. However it might be better testing more than not enough. Adding it.